### ISSUE TITLE
First test failed

### ISSUE THREAD

**Issue by wolfnether** [FreeCodeCamp/FreeCodeCamp#9274]
Challenge [Record Collection](https://www.freecodecamp.com/challenges/record-collection#?solution=%0A%2F%2F%20Setup%0Avar%20collection%20%3D%20%7B%0A%20%20%20%20%222548%22%3A%20%7B%0A%20%20%20%20%20%20%22album%22%3A%20%22Slippery%20When%20Wet%22%2C%0A%20%20%20%20%20%20%22artist%22%3A%20%22Bon%20Jovi%22%2C%0A%20%20%20%20%20%20tracks%3A%20%5B%20%0A%20%20%20%20%20%20%20%20%22Let%20It%20Rock%22%2C%20%0A%20%20%20%20%20%20%20%20%22You%20Give%20Love%20a%20Bad%20Name%22%20%0A%20%20%20%20%20%20%5D%0A%20%20%20%20%7D%2C%0A%20%20%20%20%222468%22%3A%20%7B%0A%20%20%20%20%20%20%22album%22%3A%20%221999%22%2C%0A%20%20%20%20%20%20%22artist%22%3A%20%22Prince%22%2C%0A%20%20%20%20%20%20%22tracks%22%3A%20%5B%20%0A%20%20%20%20%20%20%20%20%221999%22%2C%20%0A%20%20%20%20%20%20%20%20%22Little%20Red%20Corvette%22%20%0A%20%20%20%20%20%20%5D%0A%20%20%20%20%7D%2C%0A%20%20%20%20%221245%22%3A%20%7B%0A%20%20%20%20%20%20%22artist%22%3A%20%22Robert%20Palmer%22%2C%0A%20%20%20%20%20%20%22tracks%22%3A%20%5B%20%5D%0A%20%20%20%20%7D%2C%0A%20%20%20%20%225439%22%3A%20%7B%0A%20%20%20%20%20%20%22album%22%3A%20%22ABBA%20Gold%22%0A%20%20%20%20%7D%0A%7D%3B%0A%2F%2F%20Keep%20a%20copy%20of%20the%20collection%20for%20tests%0A%2F%2Fvar%20collectionCopy%20%3D%20JSON.parse%28stringify%28collection%29%29%3B%0A%0A%2F%2F%20Only%20change%20code%20below%20this%20line%0Afunction%20updateRecords%28id%2C%20prop%2C%20value%29%20%7B%0A%20%20if%20%28prop%20%3D%3D%20%22tracks%22%20%26%26%20value%20!%3D%3D%20%22%22%29%20%7B%0A%20%20%20%20if%20%28!collection%5Bid%5D.hasOwnProperty%28prop%29%29%0A%20%20%20%20%20%20collection%5Bid%5D%5Bprop%5D%20%3D%20%5B%5D%3B%0A%20%20%20%20%20collection%5Bid%5D%5Bprop%5D.push%28value%29%3B%0A%20%20%7D%20else%20if%20%28value%20!%3D%3D%20%22%22%29%20%7B%0A%20%20%20%20collection%5Bid%5D%5Bprop%5D%20%3D%20value%3B%0A%20%20%7D%20else%20%7B%0A%20%20%20%20delete%20collection%5Bid%5D%5Bprop%5D%3B%0A%20%20%7D%0A%20%20console.log%28id%2C%20prop%2C%20value%2Ccollection%29%3B%0A%20%20return%20collection%3B%0A%7D%0A%0A%2F%2F%20Alter%20values%20below%20to%20test%20your%20code%0AupdateRecords%285439%2C%20%22artist%22%2C%20%22ABBA%22%29%3B%0A%0A) has an issue.
User Agent is: <code>Mozilla/5.0 (Windows NT 10.0; WOW64; rv:46.0) Gecko/20100101 Firefox/46.0</code>.
Please describe how to reproduce this issue, and include links to screenshots if possible.

My code:

``` javascript
function updateRecords(id, prop, value) {
  if (prop == "tracks" && value !== "") {
    if (!collection[id].hasOwnProperty(prop))
      collection[id][prop] = [];
     collection[id][prop].push(value);
  } else if (value !== "") {
    collection[id][prop] = value;
  } else {
    delete collection[id][prop];
  }
  return collection;
}
```

**Comment by dhcodes** [comment]
@wolfnether is this issue similar to the error you are getting? https://github.com/FreeCodeCamp/FreeCodeCamp/commit/3fa86377f87c98cb5a3ea3bae84daee3424cf58c

**Comment by wolfnether** [comment]
I commented this line

**Comment by dhcodes** [comment]
Did it let you proceed @wolfnether ?

**Comment by erictleung** [comment]
@wolfnether thanks for bringing up the error. Since that change @dhcodes has pointed out, the object in the editor is *actually* a JSON object now. That line you commented out 

```js
var collectionCopy = JSON.parse(stringify(collection));
```

is important because the first test uses the `collectionCopy` variable. Commenting that out fails the first test because under the hood, it actually uses the `collectionCopy` variable, unlike the other tests. So uncomment that line out and just put

```js
var collectionCopy = collection;
```

and it should work.

cc/ @FreeCodeCamp/issue-moderators looks like we can remove [the `JSON.parse()` part in the seed code](https://github.com/FreeCodeCamp/FreeCodeCamp/blob/b4cf1037d53fc614ddfde615ada46d02ddf5eabb/seed/challenges/01-front-end-development-certification/basic-javascript.json#L4562), remove

```
"(function(x) { return \"collection = \\n\" + JSON.stringify(x, '\\n', 2); })(collection);"
```

from [the tail](https://github.com/FreeCodeCamp/FreeCodeCamp/blob/b4cf1037d53fc614ddfde615ada46d02ddf5eabb/seed/challenges/01-front-end-development-certification/basic-javascript.json#L4576) (because the object is already a JSON object), and then remove the `collection = collectionCopy` from [this line](https://github.com/FreeCodeCamp/FreeCodeCamp/blob/b4cf1037d53fc614ddfde615ada46d02ddf5eabb/seed/challenges/01-front-end-development-certification/basic-javascript.json#L4582).

**Comment by BKinahan** [comment]
This problem was recently addressed in #9276 and #9277 to revert a change that broke the challenge a bit. Once that merged code goes live these issues should no longer be a problem.

@erictleung Wouldn't `var collectionCopy = collection;` be faulty since `collection` is an object (and thus any changes to `collectionCopy` will also apply to `collection`)? I believe that is why the parse/stringify line was written as such originally, to copy the value of the object and not its reference.

**Comment by QuincyLarson** [comment]
It looks like from what @BKinahan has said that this issue is resolved. Closing as stale.
